include/config/config.h

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SYS_CONFIG_H_
+#define __SYS_CONFIG_H_
+
+#include <misc/util.h>
+#include <misc/slist.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup sys_config sys/config package
+ * @{
+ */
+
+#define CONF_MAX_DIR_DEPTH	8	/* max depth of config tree */
+#define CONF_MAX_NAME_LEN	(8 * CONF_MAX_DIR_DEPTH)
+#define CONF_MAX_VAL_LEN	256
+#define CONF_NAME_SEPARATOR	"/"
+
+#define CONF_NMGR_OP		0
+
+/**
+ * Type of configuration value.
+ */
+enum conf_type {
+	CONF_NONE = 0,
+	CONF_DIR,
+	CONF_INT8,
+	CONF_INT16,
+	CONF_INT32,
+	CONF_INT64,
+	CONF_STRING,
+	CONF_BYTES,
+	CONF_FLOAT,
+	CONF_DOUBLE,
+	CONF_BOOL,
+} __attribute__((__packed__));
+
+/**
+ * Parameter to commit handler describing where data is going to.
+ */
+enum conf_export_tgt {
+	CONF_EXPORT_PERSIST,        /* Value is to be persisted. */
+	CONF_EXPORT_SHOW            /* Value is to be displayed. */
+};
+
+/**
+ * @struct conf_handler
+ * Config handlers for subtree implement a set of handler functions.
+ * These are registered using a call to conf_register().
+ *
+ * @var conf_handler::ch_list Linked list node info for module internal usage.
+ *
+ * @var conf_handler::ch_name Name of subtree.
+ *
+ * @var conf_handler::ch_get Get values handler of configuration items
+ * identified by keyword names.Parameters:
+ *   argc - count of item in argv.
```

Comment:
This will format better as a list of items:

    identified by keyword names. Parameters:

    - argc - count of item in argv.
    - argv - array of pointers to keyword names.